Dockerfile

```diff
@@ -0,0 +1,52 @@
+FROM python:3.9-slim-buster as builder
+
+WORKDIR /usr/src/app
+
+# set environment variables
+ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONUNBUFFERED 1
+
+# install system dependencies
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends gcc
+
+# lint
+RUN python -m pip install --upgrade pip
+COPY . /usr/src/app/
+RUN python -m pip install black==23.7.0
+
+
+# install python dependencies
+COPY ./requirements.txt .
+RUN pip wheel --no-cache-dir --no-deps --wheel-dir /usr/src/app/wheels -r requirements.txt
+
+
+FROM python:3.9-slim-buster
+
+# create directory for the app user
+RUN mkdir -p /home/app
+
+# create the app user
+RUN addgroup --system app && adduser --system --group app
```

Comment:
I wouldn't deal with the user in the container unless it's absolutely necessary. Just let the root handle everything.